### DISCUSSION TITLE
Как фильтровать строки в таблице по отображаемому значению ( не по основанию ) в поле?

### DISCUSSION THREAD

**Discussion by vldekov** [totumonline/totum-ru-issues-and-discussions#417] [Как это сделать?] [answered]
Как фильтровать строки в таблице по отображаемому значению ( не по основанию ) в поле? Например, с помощью setRowsHide или префильтра.

**Comment by simplepersonru** [comment, marked as answer]
Фильтровать с точки зрения пользователя или програмно?

Если с точки зрения пользователя, то колоночным фильтром (включается в настройках поля "Фильтр по полю") https://ru.docs.totum.online/fields#fields-settings-filterable

Если программно, то можно сделать невидимую пользователю колонку, которая будет содержать непосредственно значение, а не основание. Код колонки: 
`=: select(table: '%1'; field: '%2'; where: '%3' = #%4;)`

%1 - Имя таблицы, по которой селект
%2 - Имя поля таблицы по которой селект, которое содержит нужное отображение
%3 - Имя поля-основания в таблице селекта
%4 - Имя поля из текущей таблицы, в котором лежит основание селекта

Уже по этой колонке можно делать префильтр например

Это просто способ решения в лоб, возможно есть и более изящный

**Reply by ttm-alexeypolunin** [reply]
setRowsHide тоже можно использовать — надо помнить, что это нежесткое ограничение. Он принимает id, вам нужно будет вычислить какие id нужно скрыть.
